fili-core/src/test/groovy/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandlerSpec.groovy

```diff
@@ -0,0 +1,545 @@
+        RequestContext requestContext = new RequestContext(containerRequestContext, true)
+        requestContext.isReadCache() >> true
+        MAPPER.valueToTree(groupByQuery) >> json
+        cacheService = new CacheService(dataCache, querySigningService, MAPPER)
+        CacheWeightCheckRequestHandler handler = new CacheWeightCheckRequestHandler(
+                next,
+                webService,
+                queryWeightUtil,
+                MAPPER,
+                cacheService
+        )
+
+        expect: "The count of fact query cache hit is 0"
+        bardQueryInfo.queryCounter.get(BardQueryInfo.FACT_QUERY_CACHE_HIT).get() == 0
+
+        when: "A groupBy query runs with a valid cache hit"
+        boolean requestProcessed = handler.handleRequest(requestContext, request, groupByQuery, response)
+
+        then: "Check the cache and return valid json"
+        1 * dataCache.get(_) >> new MemTupleDataCache.DataEntry<String>("key1", 1234L, "[]")
+
+        and: "The request is marked as processed"
+        requestProcessed
+
+        and: "The count of fact query cache hit is incremented by 1"
+        bardQueryInfo.queryCounter.get(BardQueryInfo.FACT_QUERY_CACHE_HIT).get() == 1
+    }
+
+    def "Test handle request on cache hit responds to the group by request with exception in cache read"() {
+        setup:
+        GroupByQuery groupByQuery = RequestUtils.buildGroupByQuery()
+        QuerySigningService querySigningService = Mock(SegmentIntervalsHashIdGenerator)
+        querySigningService.getSegmentSetId(_) >> Optional.of(1234L)
+        ContainerRequestContext containerRequestContext = Mock(ContainerRequestContext)
+        containerRequestContext.getHeaders() >> (["Bard-Testing": "###BYPASS###", "ClientId": "UI"] as
+                MultivaluedHashMap<String, String>)
+        RequestContext requestContext = new RequestContext(containerRequestContext, true)
+        requestContext.isReadCache() >> true
+        cacheService = new CacheService(dataCache, querySigningService, MAPPER)
+        CacheWeightCheckRequestHandler handler = new CacheWeightCheckRequestHandler(
+                next,
+                webService,
+                queryWeightUtil,
+                MAPPER,
+                cacheService
+        )
+
+        expect: "The count of fact query cache hit is 0"
+        bardQueryInfo.queryCounter.get(BardQueryInfo.FACT_QUERY_CACHE_HIT).get() == 0
+
+        when: "A groupBy query runs with a valid cache hit"
+        boolean requestProcessed = handler.handleRequest(requestContext, request, groupByQuery, response)
+
+        then: "Check the cache and return valid json"
+        1 * dataCache.get(_) >> new MemTupleDataCache.DataEntry<String>("key1", 1234L, "[]")
+
+        and: "The request is marked as processed"
+        requestProcessed
+
+        and: "The count of fact query cache hit is incremented by 1"
+        bardQueryInfo.queryCounter.get(BardQueryInfo.FACT_QUERY_CACHE_HIT).get() == 1
```

Comment:
This test looks good other than it would be simpler with a mocked cache service

Reply:
The cache service here would need to be built using the dataCache, and with a mocked cache service it would not follow the desired path I think.

Reply:
This is a test on a handler that you've passed a cache service into.  If you've passed a cache service into it, you shouldn't need to implement any of it's dependencies.  You can just Mock it's get and set properties.